docs/scripts/patch_devsite_toc.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+python-bigtable is made up of multiple clients, which we want to categorize separately in
+the devsite table of contents.
+
+This script allows us to add separate sections to the autogenerated toc.yml file to
+accomplish this.
```

Comment:
Please add a bit more background here:

This script will run after `nox -s docfx` is run. `docfx` is the api doc format used by google cloud. It is described here: https://github.com/googleapis/docuploader?tab=readme-ov-file#requirements-for-docfx-yaml-tarballs. 
One of the file used by docfx is toc.yml which is used to generate the table of contents sidebar. This script will patch file to create subfolders for each of the clients